src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -975,12 +975,26 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
 
                 if (multiClickMapper == 3)
                 {
-                    _terminal->MultiClickSelection(terminalPosition, ::Terminal::SelectionExpansionMode::Line);
+                    if (shiftEnabled && _terminal->IsSelectionActive())
+                    {
+                        _terminal->SetSelectionEnd(terminalPosition, ::Terminal::SelectionExpansionMode::Line);
+                    }
+                    else
+                    {
+                        _terminal->MultiClickSelection(terminalPosition, ::Terminal::SelectionExpansionMode::Line);
+                    }
                     _selectionNeedsToBeCopied = true;
                 }
                 else if (multiClickMapper == 2)
                 {
-                    _terminal->MultiClickSelection(terminalPosition, ::Terminal::SelectionExpansionMode::Word);
+                    if (shiftEnabled && _terminal->IsSelectionActive())
+                    {
+                        _terminal->SetSelectionEnd(terminalPosition, ::Terminal::SelectionExpansionMode::Word);
+                    }
+                    else
+                    {
+                        _terminal->MultiClickSelection(terminalPosition, ::Terminal::SelectionExpansionMode::Word);
+                    }

```

Comment:
Too much duplication; can this be trimmed back a bit?

Reply:
Lift out common subexpressions.

Reply:
The only thing I can think of is adding a lambda like this:
```c++
// performs a double/triple click selection
auto multiClickSelection = [this, terminalPosition, shiftEnabled](::Terminal::SelectionExpansionMode mode) {
	if (shiftEnabled && _terminal->IsSelectionActive())
	{
		_terminal->SetSelectionEnd(terminalPosition, mode);
	}
	else
	{
		_terminal->MultiClickSelection(terminalPosition, mode);
	}
	_selectionNeedsToBeCopied = true;
};
```
I'm not entirely convinced by it but it's in the latest commit for you to look at.